### PR TITLE
Add thanos sidecar to parca-analytics

### PR DESCRIPTION
**Pull request by metalmatze**
This will read the secret with the object storage config and then start sending Prometheus blocks to object storage.

### CODE REVIEW

File: monitoring/environments/civo-demo/main.jsonnet [hunk 1]
```diff
@@ -129,6 +129,12 @@ local prometheuses = [
         query+: {
           lookbackDelta: '15m',  // Analytics are only sent once every 10m
         },
+        thanos: {
+          objectStorageConfig: {
+            name: 'parca-analytics-objectstorage',
+            key: 'thanos.yaml',
+          },
```
Comment (maxbrunet) [inline monitoring/environments/civo-demo/main.jsonnet:136]: This needs to go in the function `params` after line 35.

Could you add resources limits/requests too, please?

```jsonnet
          resources+: {
            limits+: {
              memory: '64Mi',
            },
            requests+: {
              cpu: '100m',
              memory: '64Mi',
            },
          },
```

**Review by maxbrunet — CHANGES_REQUESTED** [review]
You can check if the change produces the right result with:

```shell
argocd app diff scaleway-parca-demo-monitoring --revision=parca-analytics-thanos
```

Please paste the output back here.

Also since secrets are managed by hand in the demo, could you add a note in the `monitoring/README.md` about how you create it, please?
It's useful in case we need to recreate the deployment/cluster
Here's an example: https://github.com/parca-dev/demo-deployments/tree/main/oauth2-proxy#configure-github-oauth-and-cookie-secret

I have already set up an IAM user and policy for Thanos, you can generate the API credentials here
https://console.scaleway.com/iam/applications/c8d9958c-6b90-4ef5-96a6-196870a77452/credentials

**Comment by maxbrunet** [comment]
By the way, do we really need this or simply add a PVC and retain a few days? We probably need a small PVC too anyway, otherwise we would loose recent data which hasn't cut a block yet on restart.

**Comment by brancz** [comment]
I think the long-term trend is what's most useful to understand.

**Comment by metalmatze** [comment]
Huh. I remembered these comments were somewhere. 
Seems like moving from my laptop to my desktop I somehow messed up and created an entirely new PR on https://github.com/parca-dev/demo-deployments/pull/180
Closing this one in favor of the newer one with the additional docs.

**Pull request closed without merging**